tracee-ebpf/tracee/vmlinux/x86_64/vmlinux-arch.h

```diff
@@ -0,0 +1,65 @@
+#ifndef __VMLINUX_ARCH_H__
+#define __VMLINUX_ARCH_H__
+
+#include <vmlinux-core.h>
+#include <vmlinux-flavored.h>
+
+#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
+#pragma clang attribute push (__attribute__((preserve_access_index)), apply_to = record)
+#endif
+
+struct thread_info {
+	u32 status;
+};
```

Comment:
seems that only thread_info is different between archs.
Can we put task_struct in the common header?

Reply:
note that vmlinux-arch is the one being included, not vmlinux-core (so the -I logic based on architecture could work). With that, I need task_struct declared after thread_info because of its "thread_info" field. If this grows, we can create a vmlinux-common.h and make both vmlinux-arch depend on it.

Reply:
Can't we just put thread_info before `#include <vmlinux-core.h>`?

Reply:
I also wonder why not to put this struct and pt_regs struct under ifdef ARCH, then we can have a single header instead of three

Reply:
sounds like a plan, let me re-change that (I had missed this comment earlier so please wait until your next review, I'll push the changes soon).

Reply:
what do you think about this last push @yanivagman ? I'm using a single vmlinux.h file that includes vmlinux-core.h PLUS the needed flavors.

Reply:
Looks good!
One thing that I would change is the naming:
vmlinux.h -> vmlinux-flavors.h
vmlinux-core.h -> vmlinux.h
And include vmlinux-flavors.h from vmlinux.h
WDYT?

Reply:
@yanivagman done.